reposerver/repository/repository.go

```diff
@@ -2851,6 +2895,10 @@ func (s *Service) UpdateRevisionForPaths(_ context.Context, request *apiclient.U
 	}
 	defer io.Close(closer)
 
+	if err := s.fetch(gitClient, []string{syncedRevision}); err != nil {
```

Comment:
We lock on revision above, but use syncedRevision here. Should it be consistent?

Reply:
To exec `git diff syncRevision..targetRevision` we only need to fetch from remote, so I think we needn't take a lock.

Reply:
I see. Maybe we can release the lock earlier then.